inspect - provide information about output files from input files

cscheid opened the pull request:
Closes #12805 

(@cderv, I had to fix that projectContext safeClone bug in this PR, for the same reason. Just FYI!)

cderv committed:
update to latest uv

cderv committed:
Update to python 3.13.3

cderv committed:
3.13.2 is the latest one available on Linux

cderv committed:
update uv.lock

cscheid committed:
inspect - provide information about output files from input files

cderv commented on src/project/project-context.ts:
Great! This is what we discussed indeed. 

When I worked on that after our discussion, I looked into a way to reduce duplication - it felt weird to do `clone: () => result`

So looking into what Typescript allows (still in learning phase :)), I did a factory function 
````typescript
export function projectContextFactory(
  contextProps: Omit<ProjectContext, "clone">,
): ProjectContext {
  const context = contextProps as ProjectContext;
  context.clone = () => context;
  return context;
}
````

Is this a good pattern for you? 

It would mean our code would be 
````diff
-const result: ProjectContext = {
+const result = projectContextFactory({
  resolveBrand: async (fileName?: string) =>
    projectResolveBrand(result, fileName),
  resolveFullMarkdownForFile: (
...
-}
+)};
````

cscheid replied:
This is (kind of) a good pattern, yes! As I discussed with you, the goal is to eventually make ProjectContext an actual ES6 `class` with two constructors for "multiple-file" projects and single-file projects. If we start using a factory function like this, then we will start our path to removing all the code duplication.

If we're going to do a factory function, then we should really just do it entirely, instead of having the "projectContextFactory" function mutate an existing object rather than being really a factory (ie returning a new object).

Without figuring out the TypeScript bits, this would be something like

````js
export const projectContextFactory = (option1, option2, option3, blablbal ...) => {
  const result = {
    clone: () => result;
    .. all other stuff gets put here
  }
  return result;
}
````

cderv replied:
I see, thanks a lot!

cderv commented on src/project/types/single-file/single-file.ts:
```diff
@@ -40,11 +40,12 @@ export async function singleFileProjectContext(
     dir: normalizePath(dirname(source)),
     engines: [],
     files: {
-      input: [],
+      input: [normalizePath(source)],
```
Is this to solve the relative vs absolute path problem? 

Or not directly related? I wonder if it can help 🤔

cderv reviewed (commented):
Interestingly, the cleanup test is failing... 😓 
```
./run-tests.sh ./smoke/inspect/inspect-cleanup.test.ts
```

I'll merge the other PR we discussed yesterday, and you can rebase on main to see if this change something

cscheid committed:
lua,typst - do not crash if float.content is nil

cderv committed:
Merge branch 'main' into enhance/12805

cderv reviewed (commented):
Here are the details for the difference 

This is for 
````
quarto inspect tests/docs/websites/issue-9253/index.qmd tests/docs/websites/issue-9253/index.json
````

With main we add this [index.json](https://github.com/user-attachments/files/20439788/index.json)

<details><summary>previously</summary>
<p>

````json
{
  "quarto": {
    "version": "99.9.9"
  },
  "engines": [
    "jupyter"
  ],
  "formats": {
    "html": {
      "identifier": {
        "display-name": "HTML",
        "target-format": "html",
        "base-format": "html"
      },
      "execute": {
        "fig-width": 7,
        "fig-height": 5,
        "fig-format": "retina",
        "fig-dpi": 96,
        "df-print": "default",
        "error": false,
        "eval": true,
        "cache": null,
        "freeze": false,
        "echo": true,
        "output": true,
        "warning": true,
        "include": true,
        "keep-md": false,
        "keep-ipynb": false,
        "ipynb": null,
        "enabled": null,
        "daemon": null,
        "daemon-restart": false,
        "debug": false,
        "ipynb-filters": [],
        "ipynb-shell-interactivity": null,
        "plotly-connected": true,
        "engine": "jupyter"
      },
      "render": {
        "keep-tex": false,
        "keep-typ": false,
        "keep-source": false,
        "keep-hidden": false,
        "prefer-html": false,
        "output-divs": true,
        "output-ext": "html",
        "fig-align": "default",
        "fig-pos": null,
        "fig-env": null,
        "code-fold": "none",
        "code-overflow": "scroll",
        "code-link": false,
        "code-line-numbers": false,
        "code-tools": false,
        "tbl-colwidths": "auto",
        "merge-includes": true,
        "inline-includes": false,
        "preserve-yaml": false,
        "latex-auto-mk": true,
        "latex-auto-install": true,
        "latex-clean": true,
        "latex-min-runs": 1,
        "latex-max-runs": 10,
        "latex-makeindex": "makeindex",
        "latex-makeindex-opts": [],
        "latex-tlmgr-opts": [],
        "latex-input-paths": [],
        "latex-output-dir": null,
        "link-external-icon": false,
        "link-external-newwindow": false,
        "self-contained-math": false,
        "format-resources": [],
        "notebook-links": true
      },
      "pandoc": {
        "standalone": true,
        "wrap": "none",
        "default-image-extension": "png",
        "to": "html",
        "css": [
          "styles.css"
        ],
        "toc": true,
        "output-file": "index.html"
      },
      "language": {
        "toc-title-document": "Table of contents",
        "toc-title-website": "On this page",
        "related-formats-title": "Other Formats",
        "related-notebooks-title": "Notebooks",
        "source-notebooks-prefix": "Source",
        "other-links-title": "Other Links",
        "code-links-title": "Code Links",
        "launch-dev-container-title": "Launch Dev Container",
        "launch-binder-title": "Launch Binder",
        "article-notebook-label": "Article Notebook",
        "notebook-preview-download": "Download Notebook",
        "notebook-preview-download-src": "Download Source",
        "notebook-preview-back": "Back to Article",
        "manuscript-meca-bundle": "MECA Bundle",
        "section-title-abstract": "Abstract",
        "section-title-appendices": "Appendices",
        "section-title-footnotes": "Footnotes",
        "section-title-references": "References",
        "section-title-reuse": "Reuse",
        "section-title-copyright": "Copyright",
        "section-title-citation": "Citation",
        "appendix-attribution-cite-as": "For attribution, please cite this work as:",
        "appendix-attribution-bibtex": "BibTeX citation:",
        "appendix-view-license": "View License",
        "title-block-author-single": "Author",
        "title-block-author-plural": "Authors",
        "title-block-affiliation-single": "Affiliation",
        "title-block-affiliation-plural": "Affiliations",
        "title-block-published": "Published",
        "title-block-modified": "Modified",
        "title-block-keywords": "Keywords",
        "callout-tip-title": "Tip",
        "callout-note-title": "Note",
        "callout-warning-title": "Warning",
        "callout-important-title": "Important",
        "callout-caution-title": "Caution",
        "code-summary": "Code",
        "code-tools-menu-caption": "Code",
        "code-tools-show-all-code": "Show All Code",
        "code-tools-hide-all-code": "Hide All Code",
        "code-tools-view-source": "View Source",
        "code-tools-source-code": "Source Code",
        "tools-share": "Share",
        "tools-download": "Download",
        "code-line": "Line",
        "code-lines": "Lines",
        "copy-button-tooltip": "Copy to Clipboard",
        "copy-button-tooltip-success": "Copied!",
        "repo-action-links-edit": "Edit this page",
        "repo-action-links-source": "View source",
        "repo-action-links-issue": "Report an issue",
        "back-to-top": "Back to top",
        "search-no-results-text": "No results",
        "search-matching-documents-text": "matching documents",
        "search-copy-link-title": "Copy link to search",
        "search-hide-matches-text": "Hide additional matches",
        "search-more-match-text": "more match in this document",
        "search-more-matches-text": "more matches in this document",
        "search-clear-button-title": "Clear",
        "search-text-placeholder": "",
        "search-detached-cancel-button-title": "Cancel",
        "search-submit-button-title": "Submit",
        "search-label": "Search",
        "toggle-section": "Toggle section",
        "toggle-sidebar": "Toggle sidebar navigation",
        "toggle-dark-mode": "Toggle dark mode",
        "toggle-reader-mode": "Toggle reader mode",
        "toggle-navigation": "Toggle navigation",
        "crossref-fig-title": "Figure",
        "crossref-tbl-title": "Table",
        "crossref-lst-title": "Listing",
        "crossref-thm-title": "Theorem",
        "crossref-lem-title": "Lemma",
        "crossref-cor-title": "Corollary",
        "crossref-prp-title": "Proposition",
        "crossref-cnj-title": "Conjecture",
        "crossref-def-title": "Definition",
        "crossref-exm-title": "Example",
        "crossref-exr-title": "Exercise",
        "crossref-ch-prefix": "Chapter",
        "crossref-apx-prefix": "Appendix",
        "crossref-sec-prefix": "Section",
        "crossref-eq-prefix": "Equation",
        "crossref-lof-title": "List of Figures",
        "crossref-lot-title": "List of Tables",
        "crossref-lol-title": "List of Listings",
        "environment-proof-title": "Proof",
        "environment-remark-title": "Remark",
        "environment-solution-title": "Solution",
        "listing-page-order-by": "Order By",
        "listing-page-order-by-default": "Default",
        "listing-page-order-by-date-asc": "Oldest",
        "listing-page-order-by-date-desc": "Newest",
        "listing-page-order-by-number-desc": "High to Low",
        "listing-page-order-by-number-asc": "Low to High",
        "listing-page-field-date": "Date",
        "listing-page-field-title": "Title",
        "listing-page-field-description": "Description",
        "listing-page-field-author": "Author",
        "listing-page-field-filename": "File Name",
        "listing-page-field-filemodified": "Modified",
        "listing-page-field-subtitle": "Subtitle",
        "listing-page-field-readingtime": "Reading Time",
        "listing-page-field-wordcount": "Word Count",
        "listing-page-field-categories": "Categories",
        "listing-page-minutes-compact": "{0} min",
        "listing-page-category-all": "All",
        "listing-page-no-matches": "No matching items",
        "listing-page-words": "{0} words",
        "listing-page-filter": "Filter",
        "draft": "Draft"
      },
      "metadata": {
        "lang": "en",
        "fig-responsive": true,
        "quarto-version": "99.9.9",
        "theme": "cosmo",
        "title": "Issue-9253"
      },
      "extensions": {
        "book": {
          "multiFile": true
        }
      }
    }
  },
  "resources": [
    "styles.css"
  ],
  "fileInformation": {
    "docs/websites/issue-9253/index.qmd": {
      "includeMap": [
        {
          "source": "docs/websites/issue-9253/index.qmd",
          "target": "_include.qmd"
        },
        {
          "source": "_include.qmd",
          "target": "_include2.qmd"
        },
        {
          "source": "docs/websites/issue-9253/index.qmd",
          "target": "_include.qmd"
        },
        {
          "source": "_include.qmd",
          "target": "_include2.qmd"
        }
      ],
      "codeCells": [
        {
          "start": 2,
          "end": 4,
          "file": "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\_include2.qmd",
          "source": "print(\"hello, world\")\n",
          "language": "python",
          "metadata": {}
        },
        {
          "start": 2,
          "end": 4,
          "file": "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\_include2.qmd",
          "source": "print(\"hello, world\")\n",
          "language": "python",
          "metadata": {}
        }
      ],
      "metadata": {
        "title": "Issue-9253"
      }
    }
  },
  "project": {
    "quarto": {
      "version": "99.9.9"
    },
    "dir": "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253",
    "engines": [
      "markdown",
      "jupyter"
    ],
    "config": {
      "project": {
        "type": "website",
        "lib-dir": "site_libs",
        "output-dir": "_site"
      },
      "website": {
        "title": "Issue-9253",
        "navbar": {
          "left": [
            {
              "href": "index.qmd",
              "text": "Home"
            },
            "about.qmd"
          ]
        }
      },
      "format": {
        "html": {
          "theme": "cosmo",
          "css": "styles.css",
          "toc": true
        }
      },
      "language": {}
    },
    "files": {
      "input": [
        "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\about.qmd",
        "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\index.qmd"
      ],
      "resources": [],
      "config": [
        "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\_quarto.yml"
      ],
      "configResources": [
        "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\styles.css"
      ]
    },
    "fileInformation": {
      "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\about.qmd": {
        "includeMap": [],
        "codeCells": [],
        "metadata": {
          "title": "About"
        }
      },
      "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\index.qmd": {
        "includeMap": [
          {
            "source": "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\index.qmd",
            "target": "_include.qmd"
          },
          {
            "source": "_include.qmd",
            "target": "_include2.qmd"
          },
          {
            "source": "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\index.qmd",
            "target": "_include.qmd"
          },
          {
            "source": "_include.qmd",
            "target": "_include2.qmd"
          }
        ],
        "codeCells": [
          {
            "start": 2,
            "end": 4,
            "file": "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\_include2.qmd",
            "source": "print(\"hello, world\")\n",
            "language": "python",
            "metadata": {}
          },
          {
            "start": 2,
            "end": 4,
            "file": "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\_include2.qmd",
            "source": "print(\"hello, world\")\n",
            "language": "python",
            "metadata": {}
          }
        ],
        "metadata": {
          "title": "Issue-9253"
        }
      }
    },
    "extensions": []
  }
}
````
</p>
</details> 

New one with this PR [index.json](https://github.com/user-attachments/files/20439809/index.json)

<details><summary>Now</summary>
<p>

````json
{
  "quarto": {
    "version": "99.9.9"
  },
  "engines": [
    "jupyter"
  ],
  "formats": {
    "html": {
      "identifier": {
        "display-name": "HTML",
        "target-format": "html",
        "base-format": "html"
      },
      "execute": {
        "fig-width": 7,
        "fig-height": 5,
        "fig-format": "retina",
        "fig-dpi": 96,
        "df-print": "default",
        "error": false,
        "eval": true,
        "cache": null,
        "freeze": false,
        "echo": true,
        "output": true,
        "warning": true,
        "include": true,
        "keep-md": false,
        "keep-ipynb": false,
        "ipynb": null,
        "enabled": null,
        "daemon": null,
        "daemon-restart": false,
        "debug": false,
        "ipynb-filters": [],
        "ipynb-shell-interactivity": null,
        "plotly-connected": true,
        "engine": "jupyter"
      },
      "render": {
        "keep-tex": false,
        "keep-typ": false,
        "keep-source": false,
        "keep-hidden": false,
        "prefer-html": false,
        "output-divs": true,
        "output-ext": "html",
        "fig-align": "default",
        "fig-pos": null,
        "fig-env": null,
        "code-fold": "none",
        "code-overflow": "scroll",
        "code-link": false,
        "code-line-numbers": false,
        "code-tools": false,
        "tbl-colwidths": "auto",
        "merge-includes": true,
        "inline-includes": false,
        "preserve-yaml": false,
        "latex-auto-mk": true,
        "latex-auto-install": true,
        "latex-clean": true,
        "latex-min-runs": 1,
        "latex-max-runs": 10,
        "latex-makeindex": "makeindex",
        "latex-makeindex-opts": [],
        "latex-tlmgr-opts": [],
        "latex-input-paths": [],
        "latex-output-dir": null,
        "link-external-icon": false,
        "link-external-newwindow": false,
        "self-contained-math": false,
        "format-resources": [],
        "notebook-links": true
      },
      "pandoc": {
        "standalone": true,
        "wrap": "none",
        "default-image-extension": "png",
        "to": "html",
        "css": [
          "styles.css"
        ],
        "toc": true,
        "output-file": "index.html"
      },
      "language": {
        "toc-title-document": "Table of contents",
        "toc-title-website": "On this page",
        "related-formats-title": "Other Formats",
        "related-notebooks-title": "Notebooks",
        "source-notebooks-prefix": "Source",
        "other-links-title": "Other Links",
        "code-links-title": "Code Links",
        "launch-dev-container-title": "Launch Dev Container",
        "launch-binder-title": "Launch Binder",
        "article-notebook-label": "Article Notebook",
        "notebook-preview-download": "Download Notebook",
        "notebook-preview-download-src": "Download Source",
        "notebook-preview-back": "Back to Article",
        "manuscript-meca-bundle": "MECA Bundle",
        "section-title-abstract": "Abstract",
        "section-title-appendices": "Appendices",
        "section-title-footnotes": "Footnotes",
        "section-title-references": "References",
        "section-title-reuse": "Reuse",
        "section-title-copyright": "Copyright",
        "section-title-citation": "Citation",
        "appendix-attribution-cite-as": "For attribution, please cite this work as:",
        "appendix-attribution-bibtex": "BibTeX citation:",
        "appendix-view-license": "View License",
        "title-block-author-single": "Author",
        "title-block-author-plural": "Authors",
        "title-block-affiliation-single": "Affiliation",
        "title-block-affiliation-plural": "Affiliations",
        "title-block-published": "Published",
        "title-block-modified": "Modified",
        "title-block-keywords": "Keywords",
        "callout-tip-title": "Tip",
        "callout-note-title": "Note",
        "callout-warning-title": "Warning",
        "callout-important-title": "Important",
        "callout-caution-title": "Caution",
        "code-summary": "Code",
        "code-tools-menu-caption": "Code",
        "code-tools-show-all-code": "Show All Code",
        "code-tools-hide-all-code": "Hide All Code",
        "code-tools-view-source": "View Source",
        "code-tools-source-code": "Source Code",
        "tools-share": "Share",
        "tools-download": "Download",
        "code-line": "Line",
        "code-lines": "Lines",
        "copy-button-tooltip": "Copy to Clipboard",
        "copy-button-tooltip-success": "Copied!",
        "repo-action-links-edit": "Edit this page",
        "repo-action-links-source": "View source",
        "repo-action-links-issue": "Report an issue",
        "back-to-top": "Back to top",
        "search-no-results-text": "No results",
        "search-matching-documents-text": "matching documents",
        "search-copy-link-title": "Copy link to search",
        "search-hide-matches-text": "Hide additional matches",
        "search-more-match-text": "more match in this document",
        "search-more-matches-text": "more matches in this document",
        "search-clear-button-title": "Clear",
        "search-text-placeholder": "",
        "search-detached-cancel-button-title": "Cancel",
        "search-submit-button-title": "Submit",
        "search-label": "Search",
        "toggle-section": "Toggle section",
        "toggle-sidebar": "Toggle sidebar navigation",
        "toggle-dark-mode": "Toggle dark mode",
        "toggle-reader-mode": "Toggle reader mode",
        "toggle-navigation": "Toggle navigation",
        "crossref-fig-title": "Figure",
        "crossref-tbl-title": "Table",
        "crossref-lst-title": "Listing",
        "crossref-thm-title": "Theorem",
        "crossref-lem-title": "Lemma",
        "crossref-cor-title": "Corollary",
        "crossref-prp-title": "Proposition",
        "crossref-cnj-title": "Conjecture",
        "crossref-def-title": "Definition",
        "crossref-exm-title": "Example",
        "crossref-exr-title": "Exercise",
        "crossref-ch-prefix": "Chapter",
        "crossref-apx-prefix": "Appendix",
        "crossref-sec-prefix": "Section",
        "crossref-eq-prefix": "Equation",
        "crossref-lof-title": "List of Figures",
        "crossref-lot-title": "List of Tables",
        "crossref-lol-title": "List of Listings",
        "environment-proof-title": "Proof",
        "environment-remark-title": "Remark",
        "environment-solution-title": "Solution",
        "listing-page-order-by": "Order By",
        "listing-page-order-by-default": "Default",
        "listing-page-order-by-date-asc": "Oldest",
        "listing-page-order-by-date-desc": "Newest",
        "listing-page-order-by-number-desc": "High to Low",
        "listing-page-order-by-number-asc": "Low to High",
        "listing-page-field-date": "Date",
        "listing-page-field-title": "Title",
        "listing-page-field-description": "Description",
        "listing-page-field-author": "Author",
        "listing-page-field-filename": "File Name",
        "listing-page-field-filemodified": "Modified",
        "listing-page-field-subtitle": "Subtitle",
        "listing-page-field-readingtime": "Reading Time",
        "listing-page-field-wordcount": "Word Count",
        "listing-page-field-categories": "Categories",
        "listing-page-minutes-compact": "{0} min",
        "listing-page-category-all": "All",
        "listing-page-no-matches": "No matching items",
        "listing-page-words": "{0} words",
        "listing-page-filter": "Filter",
        "draft": "Draft"
      },
      "metadata": {
        "lang": "en",
        "fig-responsive": true,
        "quarto-version": "99.9.9",
        "theme": "cosmo",
        "title": "Issue-9253"
      },
      "extensions": {
        "book": {
          "multiFile": true
        }
      }
    }
  },
  "resources": [
    "styles.css"
  ],
  "fileInformation": {
    "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\index.qmd": {
      "includeMap": [
        {
          "source": "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\index.qmd",
          "target": "_include.qmd"
        },
        {
          "source": "_include.qmd",
          "target": "_include2.qmd"
        },
        {
          "source": "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\index.qmd",
          "target": "_include.qmd"
        },
        {
          "source": "_include.qmd",
          "target": "_include2.qmd"
        }
      ],
      "codeCells": [
        {
          "start": 2,
          "end": 4,
          "file": "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\_include2.qmd",
          "source": "print(\"hello, world\")\n",
          "language": "python",
          "metadata": {}
        },
        {
          "start": 2,
          "end": 4,
          "file": "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\_include2.qmd",
          "source": "print(\"hello, world\")\n",
          "language": "python",
          "metadata": {}
        }
      ],
      "metadata": {
        "title": "Issue-9253"
      },
      "outputFiles": {
        "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\_site\\index.html": {
          "display-name": "HTML",
          "target-format": "html",
          "base-format": "html"
        }
      }
    }
  },
  "project": {
    "quarto": {
      "version": "99.9.9"
    },
    "dir": "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253",
    "engines": [
      "markdown",
      "jupyter"
    ],
    "config": {
      "project": {
        "type": "website",
        "lib-dir": "site_libs",
        "output-dir": "_site"
      },
      "website": {
        "title": "Issue-9253",
        "navbar": {
          "left": [
            {
              "href": "index.qmd",
              "text": "Home"
            },
            "about.qmd"
          ]
        }
      },
      "format": {
        "html": {
          "theme": "cosmo",
          "css": "styles.css",
          "toc": true
        }
      },
      "language": {}
    },
    "files": {
      "input": [
        "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\about.qmd",
        "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\index.qmd"
      ],
      "resources": [],
      "config": [
        "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\_quarto.yml"
      ],
      "configResources": [
        "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\styles.css"
      ]
    },
    "fileInformation": {
      "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\about.qmd": {
        "includeMap": [],
        "codeCells": [],
        "metadata": {
          "title": "About"
        },
        "outputFiles": {
          "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\_site\\about.html": {
            "display-name": "HTML",
            "target-format": "html",
            "base-format": "html"
          }
        }
      },
      "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\index.qmd": {
        "includeMap": [
          {
            "source": "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\index.qmd",
            "target": "_include.qmd"
          },
          {
            "source": "_include.qmd",
            "target": "_include2.qmd"
          },
          {
            "source": "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\index.qmd",
            "target": "_include.qmd"
          },
          {
            "source": "_include.qmd",
            "target": "_include2.qmd"
          }
        ],
        "codeCells": [
          {
            "start": 2,
            "end": 4,
            "file": "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\_include2.qmd",
            "source": "print(\"hello, world\")\n",
            "language": "python",
            "metadata": {}
          },
          {
            "start": 2,
            "end": 4,
            "file": "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\_include2.qmd",
            "source": "print(\"hello, world\")\n",
            "language": "python",
            "metadata": {}
          }
        ],
        "metadata": {
          "title": "Issue-9253"
        },
        "outputFiles": {
          "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\_site\\index.html": {
            "display-name": "HTML",
            "target-format": "html",
            "base-format": "html"
          }
        }
      }
    },
    "extensions": []
  }
}
````

</p>
</details> 

The diff shows that previously we add a mix of relative and absolute
Here are the changes where we see the outputFiles addition, but also the fileInformation key changes

````diff
❯ git diff tests\docs\websites\issue-9253\index.json
warning: in the working copy of 'tests/docs/websites/issue-9253/index.json', LF will be replaced by CRLF the next time Git touches it
diff --git a/tests/docs/websites/issue-9253/index.json b/tests/docs/websites/issue-9253/index.json
index 36c65306f..29a298fc0 100644
--- a/tests/docs/websites/issue-9253/index.json
+++ b/tests/docs/websites/issue-9253/index.json
@@ -216,10 +216,10 @@
     "styles.css"
   ],
   "fileInformation": {
-    "docs/websites/issue-9253/index.qmd": {
+    "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\index.qmd": {
       "includeMap": [
         {
-          "source": "docs/websites/issue-9253/index.qmd",
+          "source": "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\index.qmd",
           "target": "_include.qmd"
         },
         {
@@ -227,7 +227,7 @@
           "target": "_include2.qmd"
         },
         {
-          "source": "docs/websites/issue-9253/index.qmd",
+          "source": "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\index.qmd",
           "target": "_include.qmd"
         },
         {
@@ -255,6 +255,13 @@
       ],
       "metadata": {
         "title": "Issue-9253"
+      },
+      "outputFiles": {
+        "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\_site\\index.html": {
+          "display-name": "HTML",
+          "target-format": "html",
+          "base-format": "html"
+        }
       }
     }
   },
@@ -313,6 +320,13 @@
         "codeCells": [],
         "metadata": {
           "title": "About"
+        },
+        "outputFiles": {
+          "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\_site\\about.html": {
+            "display-name": "HTML",
+            "target-format": "html",
+            "base-format": "html"
+          }
         }
       },
       "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\index.qmd": {
@@ -354,6 +368,13 @@
         ],
         "metadata": {
           "title": "Issue-9253"
+        },
+        "outputFiles": {
+          "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\_site\\index.html": {
+            "display-name": "HTML",
+            "target-format": "html",
+            "base-format": "html"
+          }
         }
       }
     },
(END)
+            "base-format": "html"
+          }
         }
       },
       "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\index.qmd": {
@@ -354,6 +368,13 @@
         ],
         "metadata": {
           "title": "Issue-9253"
+        },
+        "outputFiles": {
+          "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\_site\\index.html": {
+            "display-name": "HTML",
+            "target-format": "html",
+            "base-format": "html"
+          }
         }
       }
     },
~
~
~
~
(END)
+            "base-format": "html"
+          }
         }
       },
       "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\index.qmd": {
@@ -354,6 +368,13 @@
         ],
         "metadata": {
           "title": "Issue-9253"
+        },
+        "outputFiles": {
+          "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\_site\\index.html": {
+            "display-name": "HTML",
+            "target-format": "html",
+            "base-format": "html"
+          }
         }
       }
     },
~
~
~
~
~
~
~
(END)
+            "base-format": "html"
+          }
         }
       },
       "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\index.qmd": {
@@ -354,6 +368,13 @@
         ],
         "metadata": {
           "title": "Issue-9253"
+        },
+        "outputFiles": {
+          "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\_site\\index.html": {
+            "display-name": "HTML",
+            "target-format": "html",
+            "base-format": "html"
+          }
         }
       }
     },
~
~
~
~
~
~
~
~
~
~
(END)
+            "base-format": "html"
+          }
         }
       },
       "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\index.qmd": {
@@ -354,6 +368,13 @@
         ],
         "metadata": {
           "title": "Issue-9253"
+        },
+        "outputFiles": {
+          "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\_site\\index.html": {
+            "display-name": "HTML",
+            "target-format": "html",
+            "base-format": "html"
+          }
         }
       }
     },
~
~
~
~
~
~
~
~
~
~
~
(END)
+            "base-format": "html"
+          }
         }
       },
       "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\index.qmd": {
@@ -354,6 +368,13 @@
         ],
         "metadata": {
           "title": "Issue-9253"
+        },
+        "outputFiles": {
+          "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\_site\\index.html": {
+            "display-name": "HTML",
+            "target-format": "html",
+            "base-format": "html"
+          }
         }
       }
     },
~
~
~
~
~
~
~
~
~
~
~
~
~
(END)
+            "base-format": "html"
+          }
         }
       },
       "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\index.qmd": {
@@ -354,6 +368,13 @@
         ],
         "metadata": {
           "title": "Issue-9253"
+        },
+        "outputFiles": {
+          "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\_site\\index.html": {
+            "display-name": "HTML",
+            "target-format": "html",
+            "base-format": "html"
+          }
         }
       }
     },
~
~
~
~
~
~
~
~
~
~
~
~
~
~
(END)
+            "base-format": "html"
+          }
         }
       },
       "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\index.qmd": {
@@ -354,6 +368,13 @@
         ],
         "metadata": {
           "title": "Issue-9253"
+        },
+        "outputFiles": {
+          "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\_site\\index.html": {
+            "display-name": "HTML",
+            "target-format": "html",
+            "base-format": "html"
+          }
         }
       }
     },
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
(END)
+            "base-format": "html"
+          }
         }
       },
       "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\index.qmd": {
@@ -354,6 +368,13 @@
         ],
         "metadata": {
           "title": "Issue-9253"
+        },
+        "outputFiles": {
+          "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\_site\\index.html": {
+            "display-name": "HTML",
+            "target-format": "html",
+            "base-format": "html"
+          }
         }
       }
     },
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
(END)
+            "base-format": "html"
+          }
         }
       },
       "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\index.qmd": {
@@ -354,6 +368,13 @@
         ],
         "metadata": {
           "title": "Issue-9253"
+        },
+        "outputFiles": {
+          "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\_site\\index.html": {
+            "display-name": "HTML",
+            "target-format": "html",
+            "base-format": "html"
+          }
         }
       }
     },
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
(END)
+            "base-format": "html"
+          }
         }
       },
       "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\index.qmd": {
@@ -354,6 +368,13 @@
         ],
         "metadata": {
           "title": "Issue-9253"
+        },
+        "outputFiles": {
+          "C:\\Users\\chris\\Documents\\DEV_R\\quarto-cli\\tests\\docs\\websites\\issue-9253\\_site\\index.html": {
+            "display-name": "HTML",
+            "target-format": "html",
+            "base-format": "html"
+          }
         }
       }
     },
````

cderv committed:
Merge branch 'main' into enhance/12805

cderv commented:
> I had to fix that projectContext safeClone bug in this PR, for the same reason

I made this change in a separate PR to check that our tests pass, which is the case, so I merged. 
- #12826

cderv committed:
update to latest

cderv committed:
tests - add a test for fig-format with matplotlib

cderv committed:
fix set_matplotlib_formats being removed from Ipython

the function has been deprecated from IPython 7.14 and has been removed in IPython 9.0

cderv committed:
Add to changelog

[skip ci]

cderv committed:
test - Add  a test for plotly subfigures

cderv committed:
Adapt isPlotlyLibrary regex for new plotly.py 6+ inclusion

This way the cell content is correctly detected and moved into the head

cderv committed:
Add to changelog

gordonwoodhull committed:
unified light and dark brand

fixes #12797

this introduces two brand schemas, "single" and "unified"
brand.light and brand.dark can only take single
brand takes unified and then splits it
if there are no dark customizations then the unified brand has no dark mode

cderv committed:
timing for new tests [auto PR]

MichaelHatherly committed:
Update QuartoNotebookRunner to `0.17.3`

cderv committed:
also handle non-connected notebook mode we use in dashboard

In this case, plotly.py will embed the full plotly.js library so we need to dectect this code cell and put it in the html head to not have a non figure div in the output

gordonwoodhull committed:
artifacts

snyk-bot committed:
fix: upgrade @fluentui/react-icons from 2.0.298 to 2.0.300

Snyk has created this PR to upgrade @fluentui/react-icons from 2.0.298 to 2.0.300.

See this package in npm:
@fluentui/react-icons

See this project in Snyk:
https://app.snyk.io/org/open-source-6kz/project/3a3c92ae-58ce-4c6b-b0df-2fe8c861763d?utm_source=github&utm_medium=referral&page=upgrade-pr

cderv committed:
check, knitr - Add .exe to binary on Windows for correct check of existence

github-actions[bot] committed:
Update version.txt

jkrumbiegel committed:
Don't use `copyFileSync` in case source is read-only as future writes will fail

github-actions[bot] committed:
Update version.txt

cderv committed:
timing for new tests [auto PR]

cderv committed:
Schema update - lualatex is the new default engine

rundel committed:
Add missing space with shortcode close

rundel committed:
Missing {} for multi-arg fenced div

cderv committed:
Add more verbose about python binary check and Jupyter capabilities

github-actions[bot] committed:
Update version.txt

cscheid committed:
ensure replaceAll() is called with function parameter when string is not known to be free of $ (#12853)

cscheid committed:
add regression test and changelog entry

cscheid committed:
avoid issue with \_ on pandoc attributes

cderv committed:
`--log-level critical` is supposed to be supported

cderv committed:
Add `--log-level=warn` as an alias to `--log-level=warning`

cderv committed:
Add `--log-level=debug` to the command line help

cscheid committed:
revert bad fix

cscheid committed:
work around #9224

cscheid committed:
hugo-md - add rendering for non-float layouts (#12676)

cderv committed:
Add some testing for --log-level,  --log and --log-format options

cscheid committed:
Merge branch 'main' into enhance/12805

cscheid commented on src/inspect/inspect.ts:
```diff
@@ -213,9 +216,11 @@ const inspectDocumentConfig = async (path: string) => {
       );
     }
 
+    path = normalizePath(path);
```
I added that because I wanted the output of `quarto inspect` to be independent of the directory from which it is invoked, and equivalent between single-file calls and project-wide calls. If we're going to use the JSON information to make inferences about the files, it seems that it would be more useful to not have to know which directory the call came from.

Consider a project like this:

```
find . -type f | sort
./_quarto.yml
./about.qmd
./dir1/index.qmd
./dir2/index.qmd
./index.qmd
./styles.css
```

Now consider this:

```
$ cd dir1
$ quarto inspect index.qmd | jq .fileInformation
{
  "index.qmd": {
    "includeMap": [],
    "codeCells": [],
    "metadata": {}
  }
}
$ quarto inspect index.qmd | jq .project.fileInformation
{
  "/Users/cscheid/Desktop/daily-log/2025/06/13/test-dirs/dir2/index.qmd": {
    "includeMap": [],
    "codeCells": [],
    "metadata": {}
  },
  "/Users/cscheid/Desktop/daily-log/2025/06/13/test-dirs/index.qmd": {
    "includeMap": [],
    "codeCells": [],
    "metadata": {
      "title": "test-dirs"
    }
  },
  "/Users/cscheid/Desktop/daily-log/2025/06/13/test-dirs/about.qmd": {
    "includeMap": [],
    "codeCells": [],
    "metadata": {
      "title": "About"
    }
  },
  "/Users/cscheid/Desktop/daily-log/2025/06/13/test-dirs/dir1/index.qmd": {
    "includeMap": [],
    "codeCells": [],
    "metadata": {}
  }
}
```

Notice two things:

1. We don't know which of the two `dir1/index.qmd` or `dir2/index.qmd` that first pipeline refers to.
2. The project information is _already_ built with absolute paths.

cscheid committed:
fix tests

cscheid committed:
fix tests
